eng/common/pipelines/templates/steps/publish-1es-artifact.yml

```diff
@@ -0,0 +1,34 @@
+# This step is used to prevent duplication of artifact publishes when there is an issue that would prevent the overall success of the job.
+# Ensuring that we only publish when successful (and two a differently named artifact otherwise) will allow easy retry on a build pipeline
+# without running into the "cannot override artifact" failure when we finally do get a passing run.
+
+# ArtifactName - The name of the artifact in the "successful" case.
+# ArtifactPath - The path we will be publishing.
+# CustomCondition - Used if there is additional logic necessary to prevent attempt of publish.
+
+parameters:
+  ArtifactName: ''
+  ArtifactPath: ''
+  CustomCondition: true
+  DisplayName: ''
+
+steps:
+  - task: 1ES.PublishPipelineArtifact@1
+    condition: and(succeeded(), ${{ parameters.CustomCondition }})
+    ${{ if parameters.DisplayName }}:
```

Comment:
As mentioned in the other PR I'd suggest just moving to DisplayName everywhere and for the existing places just map them to call `DisplayName: Publish ArtifactName Artifacts" if we really care about the string being the same. We could also potentially use a `coalesce` but that might not work as well for the failed condition.

Reply:
Updated a bit to remove the conditionals. I think keeping the naming for now is helpful but we can go in and remove custom ones after.

Reply:
Looks like you updated the publish-artifacts.yml file but not this one. Do you plan to keep both for the transition or did you edit the wrong one.

Reply:
Thanks for catching that, fixed.